rif-collector-app.php

```diff
@@ -34,4 +34,92 @@ function retrieveAllQuery($db) {
     return $query->fetchAll();
 }
 
-?>
+$message = "";
+$validatedSanitizedArr = [];
+
+function validateSanitizeEntry(array $entryToAdd, &$validatedSanitizedArr, &$message) {
+
+    $dateRegex = '~^\d{4}-\d{2}-\d{2}$~';
+
+    if (!is_string($entryToAdd['make'])) {
+        return $message = "Error with your 'Make' input";
+    }
+    if (!is_string($entryToAdd['model'])) {
+        return $message = "Error with your 'Model' input";
+    }
+    if (!is_string($entryToAdd['type'])) {
+        return $message = "Error with your 'Type' input";
+    }
+    if (!is_string($entryToAdd['color'])) {
+        return $message = "Error with your 'Color' input";
+    }
+    if (!filter_var($entryToAdd['mags'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Mags' input";
+    }
+    if (!is_string($entryToAdd['power'])) {
+        return $message = "Error with your 'Power' input";
+    }
+    if (!filter_var($entryToAdd['sites'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Sites Visited' input";
+    }
+    if (!preg_match($dateRegex, $entryToAdd['purchased'])) {
+        return $message = "Error with your 'Purchased Date' input";
+    } else {
+
+        $color = strip_tags($entryToAdd['color']);
+        $color = strtolower($color);
+
+        if ($color === 'white') {
+            $color = 3;
+        } elseif ($entryToAdd['color'] === 'black') {
+            $color = 2;
+        } elseif ($entryToAdd['color'] === 'tan') {
+            $color = 1;
+        } else {
+            return $message = "Error with your 'Color' input";
+        }
```

Comment:
Why does the color input accept words? Isn't it a drop down? Shouldn't it be a drop down? What if someone enters `  Black` or `white     `?

If you want it to be words, it could still be more helpful - if you only have these valid colours, your error message could help the user by telling them what the valid colours are, and what they inputted, so they can see what they put in wrong.